packages/dev/core/src/Engines/abstractEngine.ts

```diff
@@ -889,16 +889,59 @@ export abstract class AbstractEngine {
     public abstract get performanceMonitor(): PerformanceMonitor;
 
     /** @internal */
-    public _boundRenderFunction: any = () => this._renderLoop();
+    public _boundRenderFunction: any = (timestamp: number) => this._renderLoop(timestamp);
 
-    /** @internal */
-    public _renderLoop(): void {
-        // Reset the frame handler before rendering a frame to determine if a new frame has been queued.
+    protected _maxFPS: number | undefined;
+    protected _minFrameTime: number;
+    protected _lastFrameTime: number = 0;
+
+    /**
+     * Skip frame rendering but keep the frame heartbeat (begin/end frame).
+     * This is useful if you need all the plumbing but not the rendering work.
+     * (for instance when capturing a screenshot where you do not want to mix rendering to the screen and to the screenshot)
+     */
+    public skipFrameRender = false;
+
+    /** Gets or sets max frame per second allowed */
+    public get maxFPS(): number | undefined {
+        return this._maxFPS;
+    }
+
+    public set maxFPS(value: number | undefined) {

```

Comment:
Nit-picky me - if the value is negative it will never render. Maybe treat negative like `0`? i.e. - line 917 can be:
`if (value <= 0) {`

Reply:
sure :)